doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,90 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically. For example, solving $x^2 = 4$ yields $x \in \{-2,2\}$.
+
+There are two high-level functions to solve equations, [`solve`](#) and [`solveset`](#). Here are their advantages and disadvantages: ... *table?*
+
+## Using {func}`~.solve`
+
+{func}`~.solve`
+- is best for solving equations that...
+- produces outputs in the format of a Python list.
```

Comment:
Actually it produces many different output formats:
```python
In [15]: solve(x**2 - 1)
Out[15]: [-1, 1]

In [16]: solve([x**2 - 1])
Out[16]: [{x: -1}, {x: 1}]

In [17]: solve([x**2 - 1], [x])
Out[17]: [(-1,), (1,)]

In [18]: solve([x**2 - 1], [x], set=True)
Out[18]: ([x], {(-1,), (1,)})

In [19]: solve({x**2 - 1}, [x])
Out[19]: [(-1,), (1,)]

In [20]: solve({x**2 - 1}, {x})
Out[20]: [{x: -1}, {x: 1}]

In [21]: solve([x**2 - 1, y**2 - 1], [x], set=True)
Out[21]: ([x], {(-1,), (1,)})

In [22]: solve([x**2 - 1, y**2 - 1], [x, y])
Out[22]: [(-1, -1), (-1, 1), (1, -1), (1, 1)]

In [23]: solve([x**2 - 1, y**2 - 1], [x, y], dict=True)
Out[23]: [{x: -1, y: -1}, {x: -1, y: 1}, {x: 1, y: -1}, {x: 1, y: 1}]
```
The key feature of `solve` is the fact that it produces explicit solution expressions that you can use to substitute into something else. That is its purpose and that is why the list of dicts output is the best so `dict=True` should always be used to get reliably useful output:
```python
In [24]: sols = solve([x**2 - 1, y**2 - 1], [x, y], dict=True)

In [25]: expr = sin(x)*exp(y)

In [26]: expr
Out[26]: 
 y       
ℯ ⋅sin(x)

In [27]: expr.subs(sols[0])
Out[27]: 
  -1       
-ℯ  ⋅sin(1)
```

Reply:
Great, thanks. I incorporated your comment.